design.md

```diff
@@ -0,0 +1,70 @@
+# Plugin Design
+
+## Functional Requirements
+
+### Create a Project
+
+- Users can create a new project using IntelliJ's project creation dialog
+  - Projects can be libraries (no robot config file) or robots (one robot config file)
+  - A new Git repo will be initialized in the project directory
+    - A gitignore file will be added that has the relevant entries for our software, IDEs (e.g., Intellij, Eclipse, vscode), and common technologes (e.g., Java, Gradle)
+
+### Launch Robots
+
+- Users can create a run configuration that loads a robot via its config file.
+  - Users can add dev'd dependencies by repo or local file path (only supported for dependencies on other library projects).
+    - Adding via a repo should clone the repo into a local cache and then always resolve to that cache. Users can edit the repo inside the cache.
+    - Adding via a local file path simply resolves to that path. Users can edit the files under that path.
+  - Dev'ing a Maven dependency is not officially supported.
+    - ! RFC: Maven local should work with Grapes if the kernel runs on the same machine as the Maven local repository. We should investigate this. If it works, then we should officially support this case (and still not support this if the kernel is running on a different machine than the Maven local repository).
+  - Users can choose whether the kernel should connect to real hardware or to a simulator.
+    - If hardware is selected, the user must supply a connection method (e.g., USB HID, UDP).
+    - If a simulator is selected, the user must select which simulator implementation (e.g., Bullet, PhysX). The IDE must ask the kernel what simulator plugins are available.
+
+### Run Scripts
+
+- Users can create a run configuration that runs a script (this is just the default Groovy run configuration).
+  - If the user runs in the debug configuration, the IDE should automatically configure remote debugging to connect to the kernel daemon.
```

Comment:
We should explicitly note whether this includes with the kernel running on another computer. IIRC we said yes, but we should note that.